feature-libs/checkout/base/core/facade/checkout-delivery-modes.service.ts

```diff
@@ -0,0 +1,240 @@
+                   * Remove once it is switched from ngrx to c&q.
+                   * We should dispatch an event, which will load the cart$ query.
+                   */
+                  this.store.dispatch(
+                    new CartActions.LoadCart({
+                      userId,
+                      cartId,
+                    })
+                  );
+                })
+              );
+          })
+        ),
+      {
+        strategy: CommandStrategy.CancelPrevious,
+      }
+    );
+
+  protected clearDeliveryModeCommand: Command<void, unknown> =
+    this.command.create<void>(
+      () =>
+        this.checkoutPreconditions().pipe(
+          switchMap(([userId, cartId]) =>
+            this.checkoutDeliveryModesConnector
+              .clearCheckoutDeliveryMode(userId, cartId)
+              .pipe(
+                tap(() => {
+                  this.eventService.dispatch(
+                    {
+                      userId,
+                      cartId,
+                    },
+                    DeliveryModeClearedEvent
+                  );
+                  /**
+                   * TODO: We have to keep this here, since the cart feature is still ngrx-based.
+                   * Remove once it is switched from ngrx to c&q.
+                   * We should dispatch an event, which will reload the cart$ query.
+                   */
+                  this.store.dispatch(
+                    new CartActions.LoadCart({
+                      cartId,
+                      userId,
+                    })
+                  );
+                }),
+                catchError((error) => {
+                  /**
+                   * TODO: We have to keep this here, since the cart feature is still ngrx-based.
+                   * Remove once it is switched from ngrx to c&q.
+                   * We should dispatch an event, which will reload the cart$ query.
+                   */
+                  this.store.dispatch(
+                    new CartActions.LoadCart({
+                      cartId,
+                      userId,
+                    })
+                  );
+                  return throwError(error);
+                })
+              )
+          )
+        ),
+      {
+        strategy: CommandStrategy.CancelPrevious,
+      }
+    );
+
+  constructor(
+    // TODO: remove once all the occurrences are replaced with events
+    protected store: Store<StateWithMultiCart>,
+    protected activeCartService: ActiveCartService,
+    protected userIdService: UserIdService,
+    protected eventService: EventService,
+    protected query: QueryService,
+    protected command: CommandService,
+    protected checkoutDeliveryModesConnector: CheckoutDeliveryModesConnector,
+    protected checkoutQuery: CheckoutQueryFacade
+  ) {}
+
+  /**
+   * Performs the necessary checkout preconditions.
+   */
+  protected checkoutPreconditions(): Observable<[string, string]> {
```

Comment:
We actually intentionally left this fn in each of the service for extensibility purposes. The reason is that now customers can override one service, and do some specific preconditions only for that service.
If we had one common fn, it would be hard for customers to add a custom precondition checks just for one case, as it will be applied to all the usages.